crates/ra_mbe/src/mbe_expander.rs

```diff
@@ -103,6 +104,30 @@ impl Bindings {
     }
 }
 
+fn match_separator(separator: Option<char>, input: &mut TtCursor) -> Option<()> {
+    if let Some(separator) = separator {
+        if !input.is_eof() {
+            if input.eat_punct()?.char != separator {
+                return None;
+            }
+        }
+    }
+    Some(())
+}
+
+fn match_repeat(
+    subtree: &crate::Subtree,
+    res: &mut Bindings,
+    separator: Option<char>,
+    input: &mut TtCursor,
+) -> Option<()> {
+    while let Some(nested) = match_lhs(subtree, input) {
+        res.push_nested(nested)?;
+        match_separator(separator, input)?;
```

Comment:
Yeah, I think in macros separators are exactly separators, so that `$($x:ident),*` does not allow trailing commas at all.